feat: Include Rev Analytics config in cache key

rafaeelaudibert opened the pull request:
This will make us appropriately bust our queries when the revenue analytics config changes. This will bust *all* caches in the whole system, even if they aren't using Revenue Analytics data. This makes sense, however, since you can use these tables/config in any arbitrary query (Product Analytics, Web, SQL, etc.)

> [!WARNING]
> Only merge this on a weekend since we'll bust all queries' cache

rafaeelaudibert committed:
feat: Include Rev Analytics config in cache key

This will make us appropriately bust our queries when the revenue analytics config changes. This will, however, bust *all* caches in the whole system even in case they arent using Revenue Analytics data. This makes sense, however, since you can use these tables/config in any arbitrary query (Product Analytics, Web, SQL, etc.)

jabahamondes reviewed (approved):
Nice, I'll probably end up doing the same. I'd love this to happen automatically for any query detecting is any used config changes to invalidate the cache.

If you use `filter_test_accounts` you might need to add it as well there (as the bot said xd)

Gilbert09 commented on posthog/hogql_queries/test/test_query_runner.py:
```diff
@@ -113,6 +123,24 @@ def test_cache_payload(self):
+            "data_warehouse_config": {
+                "revenue_analytics": {
+                    "base_currency": "GBP",
+                    "events": [
+                        {
+                            "couponProperty": "coupon",
+                            "currencyAwareDecimal": False,
+                            "eventName": "purchase",
+                            "productProperty": "product",
+                            "revenueCurrencyProperty": {
+                                "property": "currency",
+                                "static": None,
+                            },
+                            "revenueProperty": "revenue",
+                        }
+                    ],
```
How often do/can these fields update?

rafaeelaudibert replied:
As in add new configs (keys) or change the values? New keys, very rarely, new values, everytime someone wants to add a new event source which shouldnt be very common, probably once ever? Why the concern?

rafaeelaudibert replied:
Once in a blue moon, really. There's not much reason to come back and change this after you've set it up - and it'll usually be set up when a company is small. There might be a cache busting storm when we release Rev Analytics to GA, so we need to be careful with that

rafaeelaudibert committed:
refactor: Add filter and rename key

Add the test accounts filter to the key and also use a better name (dont include dwh in the key, doesnt make a lot of sense anyway)

rafaeelaudibert commented on posthog/hogql_queries/query_runner.py:
```diff
@@ -1035,6 +1035,7 @@ def get_cache_payload(self) -> dict:
             "query": query,
             "team_id": self.team.pk,
             "hogql_modifiers": to_dict(self.modifiers),
+            "data_warehouse_config": {"revenue_analytics": self.team.revenue_analytics_config.to_cache_key_dict()},
```
Named it `product_modifiers` now

Commit:
feat: Add marketing analytics config to cache key

Included marketing analytics configuration in the cache payload to ensure accurate caching behavior. Updated tests to reflect changes in cache key generation and added a method to convert marketing analytics config to a dictionary for caching.

Commit:
feat: Introduce MARKETING_ANALYTICS_SOURCES_MAP_SAMPLE for test consistency

Added a new utility for marketing analytics sources mapping to streamline test configurations. Updated test cases to utilize this new mapping, enhancing maintainability and clarity in the test setup.

jabahamondes commented on posthog/hogql_queries/test/test_query_runner.py:
```diff
@@ -89,9 +93,32 @@ def test_init_with_query_dict(self):
+                "currency": "USD",
+                "impressions": "impressions",
+            },
+        }
```
Sounds good to me, done!

rafaeelaudibert committed:
tests: Simplify test code

github-actions[bot] committed:
Update query snapshots

jabahamondes commented:
@rafaeelaudibert the problem was the update after changing the config from Django admin, but I think this problem is bigger, I’ll take a look

Commit:
Caching marketing analytic config property

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

rafaeelaudibert committed:
refactor: Remove extra query

Because we're now caching `marketing_analytics_config` on the `team` model, we're saving a query here
